core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentGenerator.java

```diff
@@ -324,7 +325,9 @@ private void writeStartRaw(String fieldName) throws IOException {
     public void writeEndRaw() {
         assert base != null : "JsonGenerator should be of instance GeneratorBase but was: " + generator.getClass();
         if (base != null) {
-            base.getOutputContext().writeValue();
+            JsonStreamContext context = base.getOutputContext();
+            assert (context instanceof JsonWriteContext) : "Expected an instance of JsonWriteContext but was: " + context.getClass();
+            ((JsonWriteContext) context).writeValue();
```

Comment:
Not sure to get your question, but we need to ensure that we're using a `JsonWriteContext` because we need to call `writeValue()` to produce valid content (it updates some index in the context).